Packs/FeedMandiant/Integrations/FeedMandiant/FeedMandiant.py

```diff
@@ -0,0 +1,655 @@
+                       'sha1': FeedIndicatorType.File,
+                       'sha256': FeedIndicatorType.File,
+                       'url': FeedIndicatorType.URL}
+
+''' CLIENT CLASS '''
+
+
+class MandiantClient(BaseClient):
+    """Client class to interact with the service API
+    """
+
+    def __init__(self, base_url: str, username: str, password: str, verify: bool, proxy: bool, timeout: int,
+                 first_fetch: str, limit: int, types: List,
+                 metadata: bool = False, enrichment: bool = False, tags: List = [], tlp_color: Optional[str] = None):
+        super().__init__(base_url=base_url, auth=(username, password), verify=verify, proxy=proxy, ok_codes=[200])
+        self._headers = {
+            'X-App-Name': "content.xsoar.cortex.paloaltonetworks.v1.0",
+            'Accept': 'application/json',
+            'Authorization': f'Bearer {self._get_token()}'
+        }
+        self.timeout = timeout
+        self.first_fetch = first_fetch
+        self.limit = limit
+        self.types = types
+        self.metadata = metadata
+        self.tlp_color = tlp_color
+        self.tags = tags
+        self.enrichment = enrichment
+
+        LOG.add_replace_strs(self._get_token())
+
+    def _get_token(self) -> str:
+        """
+        Obtains token from integration context if available and still valid.
+        After expiration, new token are generated and stored in the integration context.
+        Returns:
+            str: token that will be added to authorization header.
+        """
+        integration_context = get_integration_context()
+        token = integration_context.get('token', '')
+        valid_until = integration_context.get('valid_until')
+
+        now_timestamp = arg_to_datetime('now').timestamp()  # type:ignore
+        # if there is a key and valid_until, and the current time is smaller than the valid until
+        # return the current token
+        if token and valid_until:
+            if now_timestamp < valid_until:
+                return token
+
+        # else generate a token and update the integration context accordingly
+        token = self._generate_token()
+
+        return token
+
+    def _generate_token(self) -> str:
+        """
+        Generates new token.
+        """
+        data = {
+            'grant_type': 'client_credentials'
+        }
+        resp = self._http_request(method='POST', url_suffix='token', resp_type='json', data=data)
+        self._token = resp.get('access_token')
+
+        integration_context = get_integration_context()
+        integration_context.update({'token': self._token})
+        # Add 10 minutes buffer for the token
+        integration_context.update({'valid_until': datetime.timestamp(datetime.now(timezone.utc)) - 600})
+        set_integration_context(integration_context)
+
+        return self._token
+
+    def get_indicator_additional_info(self, identifier: str, indicator_type: str, info_type: str = "") \
```

Comment:
Did you unify all the HTTP requests into one?

Reply:
Unified most of them.

There are 2 types of calls:

- Get indicators list - get a list of indicators by type (the basic call).
- Get indicator additional info - get additional info by indicator id.

Reply:
Not sure why did we do this it looks really confusing, but let's don't waste more time on this now.